making gatk --version print the version

lbergelson opened the pull request:
closes #5533

lbergelson commented:
```
./gatk --version
Using GATK wrapper script /Users/louisb/Workspace/gatk/build/install/gatk/bin/gatk
Running:
    /Users/louisb/Workspace/gatk/build/install/gatk/bin/gatk --version
The Genome Analysis Toolkit (GATK) v4.0.12.0-1-g10aa8c7-SNAPSHOT
```

This is different than the output of the tool --version...
```
Using GATK wrapper script /Users/louisb/Workspace/gatk/build/install/gatk/bin/gatk
Running:
    /Users/louisb/Workspace/gatk/build/install/gatk/bin/gatk PrintReads --version
Version:4.0.12.0-1-g10aa8c7-SNAPSHOT
Tool returned:
0
```

I can change this to be that version, or we can change barclay to do it differently and provide more info.  It might be useful to print the library versions with both commands as well.. that's printed during the normal run...
```
14:19:57.172 INFO  PrintReads - HTSJDK Version: 2.18.1
14:19:57.173 INFO  PrintReads - Picard Version: 2.18.16
```

@cmnbroad What do you think?

cmnbroad commented on src/main/java/org/broadinstitute/hellbender/Main.java:
```diff
@@ -338,6 +338,14 @@ private CommandLineProgram extractCommandLineProgram( final String[] args,
+                    System.out.println(String.format("%s v%s", getToolkitName(), getVersion()));
+                    return 0;
+                }
+            }.doWork();
```
It would be nice to find a solution that worked for downstream toolkits - I think this would only work if they repackaged GATK classes in their own jar ? I'm not sure I see an obvious solution that doesn't involve a lot more work though.

Either way, what do you think about adding a couple of methods to RuntimeUtils or somewhere that get the toolkit name and version number from the manifest, and then call those from here and from the corresponding CommandLineProgram methods ? It still wouldn't completely solve the downstream toolkit issue, but at least we wouldn't have to spin up a dummy CLP.  Longer term I really want to fix the special argument handling  and delegation in Barclay.

cmnbroad replied:
BTW, by "from the manifest" I meant "they way they do now", ie., getClass().getPackage().getImplementationVersion().

sooheelee commented:
This is great. For what it's worth, I would love for the Picard and HTSJDK versions to print also with `gatk --version`.

lbergelson commented:
@cmnbroad I extracted a bunch of methods to RunTimeUtils and made it cleaner.  I fixed a todo in CommandLineProgram that was related as well.  It's hard to test these things because the tests sometimes but don't always the jar with a manifest so I didn't test most of the new methods.  I can try to do something more clever if you think it's necessary though...

lbergelson commented:
resolving merge conflicts...

lbergelson commented:
@cmnbroad You're ok with this test being removed? Just wanted to double check before merging.

cmnbroad commented:
@lbergelson Yes - this test is a leftover artifact of the original implementation, but we're no longer dependent on the features tested it so its fine to remove.
